File: backend/app/infrastructure/repositories/mappers/course_mapper.py
from app.domain.entities.course import Course as CourseDomain
from app.infrastructure.repositories.orm.course import Course as CourseORM

def to_domain(orm_course: CourseORM) -> CourseDomain:
    return CourseDomain(
        id=orm_course.id,
        title=orm_course.title,
        description=orm_course.description,
        tier=orm_course.tier,
        thumbnail_url=orm_course.thumbnail_url
    )

File: frontend/lib/data/mappers/course_mapper.dart
// frontend/lib/data/mappers/course_mapper.dart
import 'package:frontend/data/models/course_model.dart';
import 'package:frontend/domain/entities/course.dart';

class CourseMapper {
  static Course fromModel(CourseModel model) {
    return Course(
      id: model.id,
      title: model.title,
      description: model.description,
      tier: model.tier,
      thumbnailUrl: model.thumbnailUrl,
      provider: model.provider,
      url: model.url,
    );
  }

  static CourseModel toModel(Course entity) {
    return CourseModel(
      id: entity.id,
      title: entity.title,
      description: entity.description,
      tier: entity.tier,
      thumbnailUrl: entity.thumbnailUrl,
      provider: entity.provider,
      url: entity.url,
    );
  }
}
